src/core/api.c

```diff
@@ -706,6 +706,7 @@ mender_api_publish_inventory_data(mender_keystore_t *inventory) {
 END:
 
     /* Release memory */
+    mender_free((void *)artifact_name);
```

Comment:
Should not `artifact_name` be defined as a non-const pointer if it is allocated on the heap?

Reply:
Why? `const` mean the data should not be changed.

Reply:
When I see `const` pointer, my initial thought is that what ever this points to, I'm not the owner of it, and I should not worry about freeing it.

When ever I see casting of a `const` pointer to a non-`const` pointer, alarm bells go off in my head.

However, my thinking may be wrong. So either way is fine by me.